src/LendController.sol

```diff
@@ -0,0 +1,156 @@
+
+    function setTermsApproval(address termsContract, bool approved) external {
+        termsApprovals[msg.sender][termsContract] = approved;
+
+        emit SetTermsApproval(msg.sender, termsContract, approved);
+    }
+
+    function purchaseNFT() external {
+        // allows anyone to call in and seize a seize NFT that
+        // the controller holds the lend ticket for and pays the lenders
+        // purchase price must be >= totalOwed on loan
+    }
+
+    function liquidateNFT(uint256 loanId) external {
+        // TBD if controller will have one liquidation mechanism 
+        // or will call out to the terms contract for implement
+        // update lenderBalance based on lenderLoanShares * sale value 
+    }
+
+    // used to lend to an exisiting loan
+    function lend(
+        uint256 loanId,
+        uint256 tokenId,
+        address nft,
+        BorrowRequest memory request,
+        bytes calldata data
+    ) external {
+        _lend(loanId, tokenId, nft, request, false, data);
+    }
+
+    function _lend(
+        uint256 loanId,
+        uint256 tokenId,
+        address nft,
+        BorrowRequest memory request,
+        bool skipIsBuyoutCheck,
+        bytes calldata data
+    ) internal {
+        LoanTerms memory terms = NFTLoanTermsSource(request.termsSource).terms(tokenId, msg.sender, request.terms.loanAsset, '');
+
+        ERC20 loanAsset = ERC20(request.terms.loanAsset);
+        uint256 lenderTotal;
+
+        for(uint i = 0; i < request.lendersSpecs.length; i++) {
+            LenderSpec memory info = request.lendersSpecs[i];
+            require(termsApprovals[info.lender][request.termsSource], 'lender has not approved terms source');
+
+            if(info.pull) {
+                ERC20(request.terms.loanAsset).safeTransferFrom(info.lender, address(this), info.amount);
+            } else {
+                lenderBalance[info.lender][address(loanAsset)] -= info.amount;
+            }
+            lenderTotal += info.amount;
+
+            lenderLoanShares[loanId][info.lender] = info.amount * ONE / terms.amount;
+        }
+
+        loanAsset.approve(address(loanFacilitator), type(uint256).max);
+
+        if(skipIsBuyoutCheck){
+            require(lenderTotal == terms.amount);
+        } else {
+            (,,, uint40 lastAccumulatedTimestamp,,,,,,,) = loanFacilitator.loanInfo(loanId);
+            
+            if(lastAccumulatedTimestamp != 0) {
+                require(terms.allowBuyouts);
+                require(lenderTotal == terms.amount + loanFacilitator.interestOwed(loanId));
+            } else {
+                require(lenderTotal == terms.amount);
+            }
+        }
+
+        loanFacilitator.lend(
+            loanId,
+            terms.interestRate,
+            terms.amount,
+            terms.durationSeconds,
+            address(this)
+        );
+    }
+
+    // used to create loan and lend to it in one transaction
+    function onERC721Received(
```

Comment:
using this onReceived hook allows us to prevent needing a multi call where users first createsLoan and then calls lend here. Although worse gas than multi call because NFT needs to go `user -> LendController -> Loan Facilitator` rather than `user -> loan facilitator`. But `loan facilitator` doesn't support multicall, anyway

Reply:
I thin it would be cool if the core backed contract, on `loanCreate` allowed the borrower to send the lend ticket somewhere via a `safeTransferFrom`, counting on an immediate loan fulfillment as the result of that transfer. So the `LendController` would receive the lend ticket, instead, check terms and just return loan payment.